src/Refactor/Xml/XmlWriter.php

```diff
@@ -0,0 +1,95 @@
+<?php
+namespace DemandwareXml\Refactor\Xml;
+
+use DemandwareXml\Refactor\Entity\WriteableEntityInteface;
+use SplFileObject;
+use XMLWriter as PhpXmlWriter;
+
+// Enhances XMLWriter with additional functionality and Demandware specific formatting.
+class XmlWriter extends PhpXmlWriter
+{
+    const NAMESPACE    = 'http://www.demandware.com/xml/impex/catalog/2006-10-31';
+    const INDENT_SPACE = ' ';
+
+    private $notEmptyWriter;
+
+    public function openFile(string $filename): bool
+    {
+        new SplFileObject($filename, 'w');
+        $this->openUri($filename);
+
+        return true;
+    }
+
+    public function setIndentDefaults(): bool
```

Comment:
We never use the return value, so this could be `void`

Reply:
@WillJW All of the code in this class is designed to mirror how `XMLWriter` functions, as it is designed to extend it with additional behaviour. Most of the methods there return a boolean, and in some cases don't have a type hint.